tokio-util/Cargo.toml

```diff
@@ -42,10 +42,11 @@ log = "0.4"
 pin-project-lite = "0.1.4"
 
 [dev-dependencies]
+lazy_static = "1.4.0"
 tokio = { version = "0.3.0", path = "../tokio", features = ["full"] }
 tokio-test = { version = "0.3.0", path = "../tokio-test" }
 
-futures = "0.3.0"
+futures = { version = "0.3.0", features = ["thread-pool"] }
```

Comment:
iirc we don't need to actually bring in this feature to test this functionality. So lets try and do it that way.

Reply:
I think we do: 
https://docs.rs/futures/0.3.5/futures/executor/struct.ThreadPool.html
> This type is only available when the thread-pool feature of this library is activated.

Reply:
We can do it via two tokio runtimes, one for the handle and another that doesn't enable the io/timer that way if you don't wrap the future with a handle it will panic but then wrapping it will work. Hopefully that makes sense.

Reply:
I added the tests. But I'm not sure how to remove the `thread-pool` feature.

Reply:
Sorry so what I am saying is doing something like:

```rust
#[test]
fn foo() {
	let rt1 = Builder::new().basic_scheduler().build().unwrap();

	let rt2 = Builder::new().basic_scheduler().enable_all().build().unwrap();

	// Failing case
    rt1.block_on(async move {
		// This should panic because we didn't enable time in this
		// runtime.
		tokio::time::delay_for(...).await.unwrap();
	});

	// Passing case
	rt1.block_on(TokioContext::new(rt2.handle().clone(), async move {
		// This will not panic because we are using the context from
		// rt2 which has `enable_all`.
		tokio::time::delay_for(...).await.unwrap();
	});
}
```

This allows us to test the functionality without needing the `ThreadPool` from the futures crate. Hopefully this makes sense, feel free to dm me on discord if you have anymore questions.